### PR TITLE
Magnus/aux bms additions

### PR DESCRIPTION
**Pull request by mglyngberg**
Added 3 new values for the aux BMS:
Strobe BMS Light
Allow Charge
Contactor Error

### CODE REVIEW

**Review by nathanaelhuh — APPROVED** [review]
Looks good to me

**Review by ExtraGravity — CHANGES_REQUESTED** [review]
The Aux BMS is a separate system than the BMS, even though it is reliant. These changes should not be categorized under just Battery I think.

**Comment by bill-luu** [comment]
Should we create a new section in the protocol for the aux battery then? Right now it's all together.

**Comment by ExtraGravity** [comment]
Yea, I think it should be separate. Every system is it's own thing conceptually in the protocol, that's how I designed it anyways. Any new systems should follow suit.

File: CommunicationLayer/MessagingFramework/MessageDefines.cpp [hunk 1]
```diff
@@ -2,9 +2,10 @@
 
 namespace
 {
-    // Message lengths do not include metadata
-    const int BATTERY_DATA_LENGTH = 51;
-    const int BATTERY_FAULTS_DATA_LENGTH = 6;
+// Message lengths do not include metadata
+    const int AUX_BMS_DATA_LENGTH = 7;
+    const int BATTERY_DATA_LENGTH = 48;
+    const int BATTERY_FAULTS_DATA_LENGTH = 7;
```
Comment (bill-luu) [inline CommunicationLayer/MessagingFramework/MessageDefines.cpp]: Why was this changed? I think this should still be 6

File: DataLayer/DataLayer.pro [hunk 2]
```diff
@@ -42,4 +44,5 @@ SOURCES += \
    KeyMotorData/KeyMotorData.cpp \
    DriverControlsData/DriverControlsData.cpp \
    MpptData/MpptData.cpp \
-   MpptData/MpptUnit.cpp
+   MpptData/MpptUnit.cpp \
+    AuxBmsData/AuxBmsData.cpp
```
Comment (bill-luu) [inline DataLayer/DataLayer.pro]: Weird indent here

File: CommunicationLayer/MessagingFramework/MessageDefines.h [hunk 3]
```diff
@@ -12,7 +12,8 @@ namespace MessageDefines
         BATTERY_FAULTS = 6,
         BATTERY = 7,
         MPPT = 9,
-        LIGHTS = 10
+        LIGHTS = 10,
+        AUX_BMS = 12
```
Comment (bill-luu) [inline CommunicationLayer/MessagingFramework/MessageDefines.h]: Actually, let's change this to 11, it doesn't make sense to leave a gap here haha

Comment (bill-luu) [reply CommunicationLayer/MessagingFramework/MessageDefines.h]: This needs to be updated on the test tool & ccs PRs as well.

**Review by bill-luu — CHANGES_REQUESTED** [review]

File: BusinessLayer/JsonMessageBuilder/JsonMessageBuilder.cpp [hunk 4]
```diff
@@ -9,11 +9,47 @@
...
 }
 
+QJsonObject JsonMessageBuilder::buildAuxBmsMessage(const I_AuxBmsData& data)
+{
+    QJsonObject auxBmsJson = QJsonObject();
+
+    switch (data.getPrechargeState())
+    {
+        case I_AuxBmsData::PrechargeState::IDLE:
+            auxBmsJson[JsonFormat::PRECHARGE_STATE] = "IDLE";
```
Comment (bill-luu) [inline BusinessLayer/JsonMessageBuilder/JsonMessageBuilder.cpp]: These are still using the old values of precharge state. We are expecting values of OFF, COMMON_ENGAGED, CHARGE_ENGAGED, ALL_ENGAGED as according to the protocol. Sorry for not catching this sooner.

**Review by bill-luu — CHANGES_REQUESTED** [review]

File: Tests/BusinessLayer/JsonMessageBuilder/JsonMessageBuilderTest.cpp [hunk 5]
```diff
@@ -1082,3 +1071,56 @@ TEST(JsonMessageBuilderTest, lights)
...
 }
+
+TEST(JsonMessageBuilderTest, auxBms)
+{
+    JsonMessageBuilder jsonMessageBuilder;
+
+    // *INDENT-OFF*
+    QString EXPECTED_JSON_MSG = "\
+        { \
+            \"PrechargeState\": \"IDLE\", \
```
Comment (bill-luu) [inline Tests/BusinessLayer/JsonMessageBuilder/JsonMessageBuilderTest.cpp]: This needs to be changed to OFF

**Review by bill-luu — CHANGES_REQUESTED** [review]
Run format as well

**Review by ExtraGravity — APPROVED** [review]
I'm just reviewing for overall architecture, looks good. I trust @bill-luu's review on all the small stuff.

**Review by bill-luu — APPROVED** [review]

**Pull request merged**